instaloader/instaloader.py

```diff
@@ -596,15 +615,15 @@ def _all_already_downloaded(path_base, is_videos_enumerated) -> bool:
                             suffix = None
                         if self.download_pictures and (not sidecar_node.is_video or self.download_video_thumbnails):
                             # pylint:disable=cell-var-from-loop
-                            filename = self.__prepare_filename(filename_template, lambda: sidecar_node.display_url)
+                            filename2 = self.__prepare_filename(filename_template, lambda: sidecar_node.display_url)
```

Comment:
I renamed this variable, because the filename is generated before the loop in [L592](https://github.com/instaloader/instaloader/blob/ec6d1f1453b8949acd989426b285fdbae579a683/instaloader/instaloader.py#L592), then for sidecar nodes the filename would be changed inside ([L618](https://github.com/instaloader/instaloader/blob/ec6d1f1453b8949acd989426b285fdbae579a683/instaloader/instaloader.py#L618), [L624](https://github.com/instaloader/instaloader/blob/ec6d1f1453b8949acd989426b285fdbae579a683/instaloader/instaloader.py#L624)) depending on which file hasn't been already downloaded and/or which slides to download, and after the loop the (post) filename is used for saving caption, metadata and so on.
So the sidecar code should use another variable and not change the post's filename, or not?

Reply:
Oh indeed. This means that when using `{filename}` in a filename pattern, the files that are created below the loop (caption txt, metadata json, etc.) are created with a wrong filename.

Could you make a PR for this change separately? Also, could we find a more descriptive name than `filename2`?

Reply:
created PR #1025